### PR TITLE
Compute editor position using selection coordinates and grid position

### PR DESCRIPTION
**Pull request by nstepien**

### CODE REVIEW

File: src/masks/InteractionMasks.tsx [hunk 1]
```diff
@@ -139,24 +132,14 @@ export default function InteractionMasks<R, K extends keyof R>({
...
-
-    const { left: portalTargetLeft, top: portalTargetTop } = editorPortalTarget.getBoundingClientRect();
-    const { scrollLeft, scrollTop } = editorPortalTarget;
+  function getEditorPosition() {
+    if (!canvas.current) return null;
+    const { left, top } = canvas.current.getBoundingClientRect();
     return {
-      left: selectionMaskLeft - portalTargetLeft + scrollLeft,
-      top: selectionMaskTop - portalTargetTop + scrollTop
+      left: columns[selectedPosition.idx].left - scrollLeft + left,
```
Comment (amanmahajan7) [inline src/masks/InteractionMasks.tsx]: Should we remove `editorPortalTarget` prop?

File: src/masks/InteractionMasks.tsx [hunk 2]
```diff
@@ -85,6 +86,7 @@ export default function InteractionMasks<R, K extends keyof R>({
   enableCellDragAndDrop,
   editorPortalTarget,
   cellNavigationMode,
+  canvas,
```
Comment (amanmahajan7) [inline src/masks/InteractionMasks.tsx]: May be we can rename it to `canvasRef` for clarity

File: src/masks/InteractionMasks.tsx (hunk 1 above)
Comment (amanmahajan7) [reply src/masks/InteractionMasks.tsx]: nm, it is needed for the portals

Comment (nstepien) [reply src/masks/InteractionMasks.tsx]: I guess we could, do we have a need for it anymore?

Comment (amanmahajan7) [reply src/masks/InteractionMasks.tsx]: I am not too sure actually 🤔

**Comment by amanmahajan7** [comment]
We can probably add a new function 

```
function getCellPosition({ idx, rowIdx }: Position) {
    if (!canvas.current) return null;
    const { left, top } = canvas.current.getBoundingClientRect();
    return {
      left: columns[idx].left - scrollLeft + left,
      top: rowIdx * rowHeight - scrollTop + top
    };
  }
```

And pass this to interaction mask. And may be use it to set `selectionMasks` dimensions also. Wdyt?

**Comment by nstepien** [comment]
We only need to get the canvas left/top when positioning elements outside the viewport though.

**Pull request merged**